File: frost-scripts/performance-test/closure-arg-bind-workload.py
#!/usr/bin/env python3

n = 1_400_000


def combine(a: int, b: int, c: int, d: int, e: int) -> int:
    return a + b + c + d + e


result = sum(combine(i, 1, 2, 3, 4) for i in range(n))
assert result == 980_013_300_000
print(result)

File: frost-scripts/performance-test/closure-arg-bind-workload.rb
#!/usr/bin/env ruby

n = 1_400_000

combine = ->(a, b, c, d, e) { a + b + c + d + e }

result = (0...n).sum { |i| combine.(i, 1, 2, 3, 4) }
raise unless result == 980_013_300_000
puts result
